### ISSUE TITLE
Missing way to set skipFormModel for OpenApi Generator

### ISSUE THREAD

**Issue by johnwc** [christianhelle/apiclientcodegen#421]
When using the `OpenApiCodeGenerator`, there is little options in tools->options. Need to be able to set other options than what is given, specifically at this time the `skipFormModel` option.

Other options available: 
- https://openapi-generator.tech/docs/customization/
- https://openapi-generator.tech/docs/generators/csharp-netcore

Can these settings be placed in the project file, to be applied per project instead of globally?

**Comment by christianhelle** [comment]
Thanks for taking the time to report this @johnwc 

It's quite easy for me to add options in the Visual Studio options pages and the settings available there are completely based on user input. I can add an option for `skipFormModel` and if you have any other flags you want to be globally configurable then any input there is greatly appreciated

As for creating project-based settings, I have been playing with the idea of introducing a configuration file for this extension that can be added to a Visual Studio project (or solution) so that you don't have to include the Swagger file (OpenAPI spec) in the project, and instead generate code based on the configuration file so it will just download the latest Swagger file on the URL configured and generate code based on that. Very similar to what the .nswag file format does. This way all options and flags used by the underlying code generator can be described there so this extension doesn't have to play catch up...

For now, I'll add the flag to the Visual Studio options page and I'll try to keep you posted on the work regarding the extension configuration file that I mentioned above

**Comment by johnwc** [comment]
I need to be able to set the `--template` argument as well.

**Comment by christianhelle** [comment]
> I need to be able to set the `--template` argument as well.

@johnwc I'm actually not familiar with this so I had to look it up, but I ended up being a bit more confused. 

What arguments would you specify for `--template`? Is it the path to the user-defined templates? If so, should this be a path relative to the Swagger spec file or should this be an absolute path for general reusable templates?

For the csharp-netcore generator I also found the `library` option which to me seems like another flavour or a template.

**Comment by johnwc** [comment]
We need to make some minor modifications to the generated classed, so we need to customize the mustache templates. The `--template` parameter allows you to point to a location that contains some or all of the mustache template files. The path should be whatever I enter in the options setting. That way if I use `./templates/` it will look for the mustache templates in a directory called templates in the same directory as the OpenAPI schema file. (Since you set the working directory of the generator to the directory path of the schema file.)

**Comment by johnwc** [comment]
BTW, the settings in the option page are not being applied after closing and reopening VS 2022. I have to go in and manually set the `Custom Additional Properties` every time I open VS.

**Comment by christianhelle** [comment]
> @christianhelle sorry, for the constant requests. But, v6.2.0 was released for this generator, can we get this updated to use the latest? The latest makes updates that allow to finally use updated version of RestSharp v108

I actually already started work on this immediately when it came out but I haven't figured out an elegant way to have backwards compatibility with .NET Framework 4.x.x. I started work on something that uses OpenAPI Generator 6.0.1 for .NET Framework projects and uses the latest on .NET Core projects.

My pace is a bit slow these days so I'll have to ask for your patience on this 😃

**Comment by johnwc** [comment]
Couldn't you add a setting in the options page to allow someone to manually set the version number? And... .NetStandard is compatible with .Net Framework, what are you referring to not being backward compatible.

**Comment by christianhelle** [comment]
> Couldn't you add a setting in the options page to allow someone to manually set the version number? And... .NetStandard is compatible with .Net Framework, what are you referring to not being backward compatible.

OpenAPI Generator 6.2.0 generates code that uses [RestSharp v108.0.1](https://www.nuget.org/packages/RestSharp/108.0.1#supportedframeworks-body-tab) which doesn't support .NET Framework, or at least I couldn't get it to work

I have a rather [comprehensive set of smoke tests](https://github.com/christianhelle/apiclientcodegen/tree/master/test/GeneratedCode/OpenApiGenerator) that I use to verify that the generated code builds on as many

I couldn't get .NET Framework projects to build with OpenAPI Generator v6.2.0 produced code and RestSharp v108.0.1

```
C:\projects\christianhelle\apiclientcodegen\test\GeneratedCode\OpenApiGenerator\Net452\Net452.csproj : error NU1202: Pa
ckage RestSharp 108.0.1 is not compatible with net452 (.NETFramework,Version=v4.5.2). Package RestSharp 108.0.1 support
s:
C:\projects\christianhelle\apiclientcodegen\test\GeneratedCode\OpenApiGenerator\Net452\Net452.csproj : error NU1202:
- net5.0 (.NETCoreApp,Version=v5.0)
C:\projects\christianhelle\apiclientcodegen\test\GeneratedCode\OpenApiGenerator\Net452\Net452.csproj : error NU1202:
- net6.0 (.NETCoreApp,Version=v6.0)
C:\projects\christianhelle\apiclientcodegen\test\GeneratedCode\OpenApiGenerator\Net452\Net452.csproj : error NU1202:
- netstandard2.0 (.NETStandard,Version=v2.0)
    0 Warning(s)
    1 Error(s)
```

I will probably end up adding an option to select which version of OpenAPI Generator to use 😄

**Comment by johnwc** [comment]
You don't have to use RestSharp for the generator, you can tell it to use HttpClient or generichost library. The default is RestSharp, that is why the setting for library in the options page is a good need.

**Comment by christianhelle** [comment]
> You don't have to use RestSharp for the generator, you can tell it to use HttpClient or generichost library. The default is RestSharp, that is why the setting for library in the options page is a good need. And... I think it is ok to drop testing if 4.5.2 still works, it is EoL already. For those using such an outdated framework they need to update or look at using the tools manually via CLI.

That is a very good point. Although .NET Framework isn't entirely dead yet, in fact [v4.8.1](https://dotnet.microsoft.com/en-us/download/dotnet-framework) came out just a few months ago

I haven't tried yet setting the `library` option yet and I'm very curious on how they use HttpClient or the generic host

**Comment by johnwc** [comment]
And v4.8.1 is supported by restsharp v108 😉

**Comment by christianhelle** [comment]
@johnwc By the way, I just merged in support for OpenAPI Generator v6.2.0 (#422)

**Comment by christianhelle** [comment]
> BTW, the settings in the option page are not being applied after closing and reopening VS 2022. I have to go in and manually set the `Custom Additional Properties` every time I open VS.

I missed this comment and can confirm that you're right. Its very odd but I'm guessing the underlying tooling which has a ton of legacy attached to it doesn't like nullable strings. Hopefully this is a quick fix!
